### PR TITLE
Yuhsuan/1484 cursor widget

### PR DESCRIPTION
**Pull request by YuHsuan-Hwang**
This PR is for the first part of https://github.com/CARTAvis/carta-frontend/issues/1484: add basic UI and show cursor info of the active frame.
The second part (multi image cursor info) will be implemented in a separate PR.

### CODE REVIEW

**Comment by kswang1029** [comment]
We could consider to show the pixel unit in a new line to save space with a narrower default column width
Screen Shot 2021-07-12 at 1 14 17 PM

**Comment by kswang1029** [comment]
We should show Stokes string instead of a numeric index. 

Screen Shot 2021-07-12 at 1 18 33 PM

**Comment by YuHsuan-Hwang** [comment]
> We could consider to show the pixel unit in a new line to save space with a narrower default column width
> Screen Shot 2021-07-12 at 1 14 17 PM

Ok. I will try if this saves more space.

**Comment by YuHsuan-Hwang** [comment]
> We should show Stokes string instead of a numeric index.
> 
> Screen Shot 2021-07-12 at 1 18 33 PM

I followed the stokes column in the image list widget. Maybe we open a new issue for this?

**Comment by kswang1029** [comment]
> > We should show Stokes string instead of a numeric index.
> > Screen Shot 2021-07-12 at 1 18 33 PM
> 
> I followed the stokes column in the image list widget. Maybe we open a new issue for this?

ok https://github.com/CARTAvis/carta-frontend/issues/1562

**Comment by YuHsuan-Hwang** [comment]
image
@kswang1029 Tried Value (140px) -> Value (90px) and Unit (60px). I think it's not saving space because of padding, so I prefer one column.

**Comment by kswang1029** [comment]
> image
> @kswang1029 Tried Value (140px) -> Value (90px) and Unit (60px). I think it's not saving space because of padding, so I prefer one column.

Oh what I mean is that we use one column but have the unit in a new line

**Comment by YuHsuan-Hwang** [comment]
> > image
> > @kswang1029 Tried Value (140px) -> Value (90px) and Unit (60px). I think it's not saving space because of padding, so I prefer one column.
> 
> Oh what I mean is that we use one column but have the unit in a new line

@kswang1029 I see... Changed to two lines

**Review by kswang1029 — APPROVED** [review]

File: src/components/CursorInfo/CursorInfoComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        imageCoords[activeFrameIndex] = (
+            <React.Fragment>
+                {toFixed(frame.cursorInfo?.posImageSpace?.x, 3)}
+                <br />
+                {toFixed(frame.cursorInfo?.posImageSpace?.y, 3)}
+            </React.Fragment>
+        );
+
+        let columnsData = new Map<number, any>();
+        columnsData.set(0, {dataType, data: imageNames});
```
Comment (veggiesaurus) [inline src/components/CursorInfo/CursorInfoComponent.tsx]: elsewhere in the codebase we use the `Map` array constructor (e.g. `new Map([[key1, val1], [key2, val2]])`. I'm not sure if that would be neater or not 🤔

File: src/components/Shared/Tables/SimpleTableComponent.tsx [hunk 2]
```diff
@@ -1,20 +1,36 @@
 import * as React from "react";
 import {observer} from "mobx-react";
-import {Cell, Column, Table, SelectionModes, RenderMode} from "@blueprintjs/table";
+import {Cell, Column, Table, SelectionModes, RenderMode, RowHeaderCell} from "@blueprintjs/table";
 import {CARTA} from "carta-protobuf";
-import {ProcessedColumnData} from "models";
+//import {ProcessedColumnData} from "models";
```
Comment (veggiesaurus) [inline src/components/Shared/Tables/SimpleTableComponent.tsx]: remove comment

File: src/components/Shared/Tables/SimpleTableComponent.tsx [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
+    defaultRowHeight?: number;
+    enableGhostCells?: boolean;
+    isIndexZero?: boolean;
     updateTableRef?: (ref: Table) => void;
 }
 
 @observer
 export class SimpleTableComponent extends React.Component<SimpleTableComponentProps> {
-    private renderDataColumn(columnName: string, columnData: any) {
+    private widths = this.props.defaultColumnWidths;
```
Comment (veggiesaurus) [inline src/components/Shared/Tables/SimpleTableComponent.tsx]: does this work properly? shouldn't this be done in `constructor`?

**Review by veggiesaurus — CHANGES_REQUESTED** [review]
couple of minor queries

File: src/components/CursorInfo/CursorInfoComponent.tsx (hunk 1 above)
Comment (YuHsuan-Hwang) [reply src/components/CursorInfo/CursorInfoComponent.tsx]: I see. I think it's neater to use `new Map([[key1, val1], [key2, val2]])`. I followed code in https://github.com/CARTAvis/carta-frontend/blob/dev/src/stores/FileBrowserStore.ts#L531. Maybe I also change these lines?

File: src/components/Shared/Tables/SimpleTableComponent.tsx (hunk 3 above)
Comment (YuHsuan-Hwang) [reply src/components/Shared/Tables/SimpleTableComponent.tsx]: I think both works, but more straightforward to initialize it in constructor. I will adjust this part.

File: src/components/CursorInfo/CursorInfoComponent.tsx (hunk 1 above)
Comment (veggiesaurus) [reply src/components/CursorInfo/CursorInfoComponent.tsx]: Yes please 😊

**Comment by YuHsuan-Hwang** [comment]
@kswang1029 @veggiesaurus @duidae 
Several new changes:
* Show one line in the value column if there's no unit
* Avoid the column widths changing when switching tabs: moved the variables from SimpleTableComponent.tsx to CursorInfoComponent.tsx.
* Avoid catalog header in the import catalog dialog crashing
* Add the widget to layout config

File: src/components/Shared/Tables/SimpleTableComponent.tsx (hunk 3 above)
Comment (YuHsuan-Hwang) [reply src/components/Shared/Tables/SimpleTableComponent.tsx]: moved the variables to CursorInfoComponent.tsx so the issue no longer exists

File: src/components/CursorInfo/CursorInfoComponent.tsx (hunk 1 above)
Comment (YuHsuan-Hwang) [reply src/components/CursorInfo/CursorInfoComponent.tsx]: Changed both parts

**Comment by kswang1029** [comment]
Two suggestions: 
1) please add a placeholder for the in-app help manual (in this PR please)
2) consider to use bold face text for the active image just like the image list widget (in this or next PR)

**Comment by YuHsuan-Hwang** [comment]
> Two suggestions:
> 
> 1. please add a placeholder for the in-app help manual (in this PR please)
> 2. consider to use bold face text for the active image just like the image list widget (in this or next PR)

1. Ok. I will add a placeholder 2. I think we can leave it to the next PR

**Review by veggiesaurus — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
@kswang1029 @veggiesaurus Added a placeholder for help manual

**Pull request merged**